jerry-math/include/math.h

```diff
@@ -22,8 +22,8 @@ extern "C"
 #endif /* __cplusplus */
 
 /* General Constants. */
-#define INFINITY    (1.0/0.0)
-#define NAN         (0.0/0.0)
+#define INFINITY    (1e+300 * 1e+300) /* 1e+300*1e+300 must overflow */
+#define NAN         (INFINITY * 0.0)
```

Comment:
So, I did my homework. And it just turned out that (a) I did not know precisely how the standard specifies infinity and NaN, and (b) jerry-math's math.h was incorrect all along.

According to the specs:
- `INFINITY` and `NAN` are _float_ constant expressions, and
- `HUGE_VAL` is a _double_ constant expression (that evaluates to infinity).

Therefore, I suggest to have these macros as suggested _for the Windows/MSVC configuration_, but also keep a slightly updated version of the existing macros for other configs. That is:
```c
#ifdef _MSC_VER
#define INFINITY    ((float) (1e+300 * 1e+300))
#define NAN         ((float) (INFINITY * 0.0f))
#else
#define INFINITY    ((float) (1.0/0.0))
#define NAN         ((float) (0.0/0.0))
#endif
#define HUGE_VAL    ((double) INFINITY)
```